impl/src/main/java/org/ehcache/impl/internal/store/tiering/TieredStore.java

```diff
@@ -84,35 +81,21 @@ public void invalidateAllWithHash(long hash) throws StoreAccessException {
     StatisticsManager.associate(authoritativeTier).withParent(this);
   }
 
-
   @Override
   public ValueHolder<V> get(final K key) throws StoreAccessException {
     try {
       return cachingTier().getOrComputeIfAbsent(key, keyParam -> {
         try {
           return authoritativeTier.getAndFault(keyParam);
         } catch (StoreAccessException cae) {
-          throw new ComputationException(cae);
+          throw new StorePassThroughException(cae);
         }
       });
-    } catch (ComputationException ce) {
-      throw ce.getStoreAccessException();
-    }
-  }
-
-  static class ComputationException extends RuntimeException {
-
-    public ComputationException(StoreAccessException cause) {
-      super(cause);
-    }
-
-    public StoreAccessException getStoreAccessException() {
-      return (StoreAccessException) getCause();
-    }
-
-    @Override
-    public synchronized Throwable fillInStackTrace() {
-      return this;
```

Comment:
Slightly unrelated, but we should probably do the same trick in `StorePassThroughException`

Reply:
I guess you are right since it disappears out of the store